tests/end_to_end/test_account.py

```diff
@@ -17,15 +18,36 @@ async def class_hashes():
 
 
 @pytest_asyncio.fixture(scope="function")
-async def erc20_token(deploy_solidity_contract, new_account):
+async def counter(deploy_solidity_contract, new_account):
     return await deploy_solidity_contract(
-        "UniswapV2",
-        "ERC20",
-        TOTAL_SUPPLY,
+        "PlainOpcodes",
+        "Counter",
         caller_eoa=new_account.starknet_contract,
     )
 
 
+@pytest.fixture(autouse=True)
+async def cleanup(invoke, class_hashes):
+    yield
+    await invoke(
+        "kakarot",
+        "set_account_contract_class_hash",
+        class_hashes["account_contract"],
+    )
+    await invoke(
+        "kakarot", "set_cairo1_helpers_class_hash", class_hashes["Cairo1Helpers"]
+    )
+
+
+async def assert_counter_transaction_success(counter, new_account):
+    """
+    Assert that the transaction sent, other than upgrading the account contract, is successful.
+    """
+    assert await counter.count() == 0
```

Comment:
you do prev_count, inc, new_count == prev_count + 1 and set counter with scope="module" to deploy it only once